packages/root/src/Root.spec.tsx

```diff
@@ -1,77 +1,133 @@
 import React from 'react';
+import '@testing-library/jest-dom/extend-expect';
 import { render, cleanup } from '@testing-library/react';
 import Root from '.';
+import { RootProps } from './Root';
 
-afterAll(cleanup);
+// types
+interface SharedProps {
+  name?: string;
+}
 
-describe('packages/root-hi', () => {
-  const props = { name: 'testName' };
-  const content = 'button content';
+interface RenderedEls {
+  root?: HTMLElement | null;
+  child?: HTMLElement | null;
+}
+
+interface LinkWrapperProps {
+  href: string;
+  target: string;
+  children: React.ReactNode;
+}
+
+type testProps = RootProps<SharedProps>;
+type unmountQueueType = Array<() => boolean>;
+
+describe('packages/root', () => {
+  const unmountQueue: unmountQueueType = [];
+  const renderedEls: RenderedEls = {};
+  const linkWrapperFn = jest.fn();
+
+  function LinkWrapper({ href, target, children, ...rest }: LinkWrapperProps) {
+    linkWrapperFn();
```

Comment:
This is a subtle but to me important, to show that in fact we actually did render the component properly (because `linkWrapperFn` will wind up getting called in the process), rather than only checking for the output.